src/components/DistanceRequest/index.js

```diff
@@ -151,6 +182,15 @@ function DistanceRequest({transactionID, report, transaction, route, isEditingRe
         [transactionID, waypoints, waypointsList],
     );
 
+    const submitWaypoints = useCallback(() => {
+        // If there is any error or loading state, don't let user go to next page.
+        if (_.size(validatedWaypoints) < 2 || hasRouteError || isLoadingRoute || isLoading) {
```

Comment:
This condition should have taken into account if the user was offline or not. This later caused https://github.com/Expensify/App/issues/32397